tests/typed_ir.tests.config

```diff
@@ -4,4 +4,3 @@ prelude=tests/empty_prelude.links
 optimise=false
 generalise_toplevel=true
 show_quantifiers=true
-recheck_frontend_transformations=true
```

Comment:
Could the recheck flag be disabled just for the new test #871?  I think in general it is good to do the revalidation for the typed IR tests, so that IR typing failures that arise because of a bug in an earlier desugaring stage are correctly attributed.  (On the other hand, this situation also illustrates that revalidation can mask problems that would otherwise be caught by IR typechecking, because revalidation is currently side-effecting.  So maybe we should be checking the typed IR tests both with and without revalidation.)

Reply:
Done now.